app/license.go

```diff
@@ -5,21 +5,33 @@ package app
 
 import (
 	"bytes"
-	"errors"
 	"net/http"
 	"os"
 	"strings"
+	"time"
 
+	"github.com/dgrijalva/jwt-go"
 	"github.com/mattermost/mattermost-server/v5/mlog"
 	"github.com/mattermost/mattermost-server/v5/model"
 	"github.com/mattermost/mattermost-server/v5/utils"
+
+	"github.com/pkg/errors"
 )
 
 const (
-	requestTrialURL = "https://customers.mattermost.com/api/v1/trials"
-	LicenseEnv      = "MM_LICENSE"
+	requestTrialURL           = "https://customers.mattermost.com/api/v1/trials"
+	LicenseEnv                = "MM_LICENSE"
+	LicenseRenewalURL         = "https://customers.mattermost.com/subscribe/renew"
+	JWTDefaultTokenExpiration = 7 * 24 * time.Hour // 7 days of expiration
```

Comment:
@jasonblais heads up that the license renewal links have a validity of 7 days and will not be valid after this timeframe for security. Customer Support/Sales can generate another if the customer didn't perform the renewal within that timeframe. 

Open to suggestions on the amount of time.

Reply:
The token will be available for an amount of time, as Maria said, but there is no problem to go into de system console and generate a new one. 

From my PoV this amount should be smaller, like 1 day for example but I know the enterprise times for renewal could be greater than expected.

Reply:
@marianunez @ethervoid At what point is the renewal link generated, and is a new one automatically created once the previous link expires?

Reply:
@jasonblais is not automatic. The user should go to the screen and generate a new token

Reply:
@ethervoid I tried to create a test server to see the screen, but looks like it's not created at the moment. Would you be open to sharing a screenshot of what it looks like?

As a first reaction, this doesn't seem obvious from the user's perspective. I think most admins are going to click on the 'Renew license now' button in the header and be confused why the link doesn't work.

![image](https://user-images.githubusercontent.com/13119842/102231886-799c6b00-3ebc-11eb-98ba-925164ebeb90.png)

Is there a possibility where after 7 days, if an admin clicks on the `renew license now` button, we auto-generate a new token for them, then redirect the admin to the renewal page? This would make the experience faster and more obvious, following our [design principles](https://docs.mattermost.com/developer/fx-guidelines.html#fast-obvious-forgiving). We could also add a note somewhere on the renewal page saying something like `This link is only valid for 7 days.`.

Let me know what you think.

cc @alexsiclari as well for thoughts.

Reply:
@jasonblais as soon as the user clicks that button, the link will be valid because we will call the endpoint added in this PR which will always return a valid url (if it generated one before and it's expired it will regenerate a new one).

The expiration affects mostly when the link is handed out directly, either by email by us directly (i.e the case for people that have not upgraded 5.32) or if we offer someway they can copy the link for airgapped cases (not scoped for V1). 

In the case they go to an expired link, we need to show them an appropriate error message in the customer portal to request a new link.

Reply:
@jasonblais This PR is only the server side logic, we're not including yet the frontend/webapp part here.

For the rest of the questions, is what @marianunez said

Reply:
Great, that sounds good to me. 7 days is fine in this use case - I'll check with Jeff Johnson as well, but no need to wait for that.

> In the case they go to an expired link, we need to show them an appropriate error message in the customer portal to request a new link.

Yes, please work with Alex Siclari on the design. We may also want to add a `This link is only valid for 7 days.` on the customer portal renewal page, but I'm 0/5 on that.

Reply:
Yes, let me know what you need. I will accommodate the text in some way.

Reply:
@ethervoid @marianunez confirmed with Jeff that 7-day expiry is fine. UX design is being discussed in the Customer Portal channel.

Resolving this comment.